Update rules for a change in `T*` substitution. (#633)
jspecify/jspecify: docs/docs/spec.md

    > This is the second easy case: `A` never includes `null`.

-   `A` has a [nullness-subtype-establishing path] to any type whose base type
    is the same as the base type of `F`, and there is *not* reason to be

>     [worried] that `F` has [nullness operator] `MINUS_NULL`.

Reviewer: Should we add an unspecified type example to the non-normative example below?

Author: I see that we don't even have a `MINUS_NULL` example right now. I've added one and then said some things about unspecified nullness. I'm not sure if I picked good things to say or not.

    > `ArrayList` may be instantiated as either an `ArrayList<@Nullable String>`
    > or an `ArrayList<String>`.)
    >
    > Subtyping questions for type-variable usages are more complex. For
    > example:
    >
    > -   `E` is a nullness subtype of `E`; `@Nullable E` is not.
    > -   Similarly, if `<F extends E>` (in null-marked code), then `F` is a
    >     nullness subtype of `E`. But if `<F extends @Nullable E>`, it is not.
    > -   `E` is a nullness subtype of `E` but not of `@NonNull E`.
    >
    > When some types have unspecified nullness, the rules become more complex
    > still:
    >
    > -   A declaration like `<F extends E>` might or might not be "intended" to

>     >     be `<F extends @Nullable E>`. Depending on what was indended, `E`
>     >     *might* be intended to be a nullness subtype of `F`.

Reviewer: I _think_ you meant this?

```suggestion
    >     be `<F extends @Nullable E>`. Depending on what was indended, `F`
    >     *might* be intended to be a nullness subtype of `E`.
```

    > -   Or that declaration might be "intended" to be `<F extends @NonNull

>     >     E>`. In that [world], `F` would be not only a nullness subtype of `F`

Reviewer: ```suggestion
    >     E>`. In that [world], `F` would be not only a nullness subtype of `E`
```

The output of the process is an augmented type.

> To determine the output, apply the following rules in order. Once one condition
> is met, skip the remaining conditions.
> 
> -   If the nullness operator to apply is `NO_CHANGE`, then the output augmented
>     type is the input augmented type.
> 
> -   If the input augmented type is an [intersection type], then the output is
>     also an intersection type. For every element `Tᵢ` of the input type, the
>     output type has an element that is the output of applying the desired
>     nullness operator to `Tᵢ`.
> 
> -   The output is a type that is the same as the input augmented type except
>     with its nullness operator set to the nullness operator to apply.

Reviewer: ```suggestion
To determine the output, apply the following rules in order.

-   If the nullness operator to apply is `NO_CHANGE`, then the output augmented
    type is the input augmented type.

-   Otherwise, if the input augmented type is an [intersection type], then the output is
    also an intersection type. For every element `Tᵢ` of the input type, the
    output type has an element that is the output of applying the desired
    nullness operator to `Tᵢ`.

-   Otherwise, the output is a type that is the same as the input augmented type except
    with its nullness operator set to the nullness operator to apply.
```

Is this more direct?

Author: Probably. We have a couple other "skip the remaining conditions" sections that you're welcome to touch up if you'd like.
